### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Fixes misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/fslang-design/actions/runs/9449892878#summary-26027243704

The action will report that the changes in this PR would make it happy: https://github.com/jsoref/fslang-design/actions/runs/9449900677#summary-26027271681

### CODE REVIEW

File: FSharp-4.0/ClassNamesAsFunctionsDesignAndSpec.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ thus generating a type ty, and process the object construction (fun v -> new ty(
 Particular care is needed for the cases where the type name 
 is a generic type (or one of a number of generic types overloaded by generic arity).
 
-The IDE tests need care, since there are some code paths that may potentially affect the intellisence results given for cases like
+The IDE tests need care, since there are some code paths that may potentially affect the IntelliSense results given for cases like
```
Comment (jsoref) [inline FSharp-4.0/ClassNamesAsFunctionsDesignAndSpec.md:62]: brand

File: FSharp-4.1/FS-1006-struct-tuples.md [hunk 2]
```diff
@@ -294,7 +294,7 @@ is
 ```
 
 # Examples and Error Messages
-[exampes]: #examples
```
Comment (jsoref) [inline FSharp-4.1/FS-1006-struct-tuples.md:297]: ?

File: FSharp-4.1/FS-1013-enable-reflection-functionality-on-portable-profiles.md [hunk 3]
```diff
@@ -52,7 +52,7 @@ type FSharpType with
     static member IsExceptionRepresentation: exceptionType:Type * ?nonPublic:bool -> bool
 ```
 
-The user mustwould opt-in to by using
```
Comment (jsoref) [inline FSharp-4.1/FS-1013-enable-reflection-functionality-on-portable-profiles.md:55]: possibly `must/would`

File: FSharp-5.0/FS-1001-StringInterpolation.md [hunk 4]
```diff
@@ -6,11 +6,11 @@ There is an approved-in-principle [proposal](http://fslang.uservoice.com/forums/
 * [Discussion](https://github.com/fsharp/fslang-design/issues/368)
 * [x] Implementation: [first draft by @yatli](https://github.com/dotnet/fsharp/pull/8966), [final merged to master (--langversion:preview)](https://github.com/dotnet/fsharp/pull/8907)
 * [x] Design review meeting (26/06/2020, online, @dsyme, @cartermp, @TIHan, @jonsequitur), [notes](https://github.com/dotnet/fsharp/pull/8907#issuecomment-650304311)
-* [x] Implementation review meeting with @cartermap and @TIHan, [notes](https://gist.github.com/dsyme/dc86bf86de81b83b75557d4944db43c2).
+* [x] Implementation review meeting with @cartermp and @TIHan, [notes](https://gist.github.com/dsyme/dc86bf86de81b83b75557d4944db43c2).
```
Comment (jsoref) [inline FSharp-5.0/FS-1001-StringInterpolation.md:9]: @cartermp is misspelled in a couple of ways throughout this repository...

File: FSharp-6.0/FS-1087-resumable-code.md [hunk 5]
```diff
@@ -1024,7 +1024,7 @@ Next we define the `Run` method for the builder.
                     cr.Machine <- sm
                     cr :> Coroutine))
         else 
-            failwith "dynamic implementation nyi"
+            failwith "dynamic implementation NYI"
```
Comment (jsoref) [inline FSharp-6.0/FS-1087-resumable-code.md:1027]: I'm assuming this is `Not-Yet-Implemented`

File: FSharp-8.0/FS-1136-constraint-intersection-syntax.md [hunk 6]
```diff
@@ -33,7 +33,7 @@ The following snippet lists examples of the current and proposed ways of using s
 ```fsharp
 let f1old (x: 't when 't :> IDisposable and 't :> int seq) = ()
 let f1new1 (x: 't & #IDisposable & #(int seq)) = () 
-// ` 't & ` may be omitted when when do not need to refer back to the type parameter
+// ` 't & ` may be omitted when they do not need to refer back to the type parameter
```
Comment (jsoref) [inline FSharp-8.0/FS-1136-constraint-intersection-syntax.md:36]: this is an atypical correction -- normally I just drop the duplicated word.

File: FSharp.Core-4.5.0.0/FS-1057-valueoption.md [hunk 7]
```diff
@@ -12,14 +12,14 @@ Likewise, the idea of a struct option type is implied by this suggestion and has
...
-This RFC covers Part 1 of [RFC FS-1039 Struct representation for active patterns](../FSharp-6.0/FS-1039-struct-representation-for-active-patterns.md), but factors out the specific conern of a struct-based optional type, `ValueOption`.
+This RFC covers Part 1 of [RFC FS-1039 Struct representation for active patterns](../FSharp-6.0/FS-1039-struct-representation-for-active-patterns.md), but factors out the specific concern of a struct-based optional type, `ValueOption`.
 
 # Motivation
 [motivation]: #motivation
 
 Powerful and extensible features such as Active Patterns and Optional Arguments should be near-zero-cost abstractions. A `ValueOption` type is a first step in this direction.
 
-Additionally, in situations where a struct can provide a benefit over a reference type, `ValueOption` may allow developers to eek out more performance independently of their usage of Active Patterns and Optional Arguments.
+Additionally, in situations where a struct can provide a benefit over a reference type, `ValueOption` may allow developers to seek out more performance independently of their usage of Active Patterns and Optional Arguments.
```
Comment (jsoref) [inline FSharp.Core-4.5.0.0/FS-1057-valueoption.md:22]: I'm not certain about this. I could be wrong...

If `eek` was indeed intended, I think using some other word might be better...

File: FSharp.Core-4.5.0.0/FS-1059-improve-async-stack-traces.md [hunk 8]
```diff
@@ -82,7 +82,7 @@ Here is the stack:
 
 Barely any user code is on the stack at all, and the line numbers are all closures in the implementation of the DSL.
 
-It is also important to note that C# async and other implementations of Promises/Tasks/"Hot Tasks" don't suffer this problem since they are not "delayef". But most F# DSLs are "delayed". For `async { }`, it is F#'s use of "cold tasks" that require explicit starting that causes the issue.
```
Comment (jsoref) [inline FSharp.Core-4.5.0.0/FS-1059-improve-async-stack-traces.md:85]: it's possible this was intended...

File: archived/FS-1061-Allow-intrinsic-type-extensions-for-provided-types.md [hunk 9]
```diff
@@ -171,7 +171,7 @@ The only drawback is the complexity of the addition, but the addition makes prov
 # Alternatives
 [alternatives]: #alternatives
 
-The alternative is to add features to the provided type by alternative means such as inheriting from the provided type or by augmenting another type which defers to the provided type internally.  These drawbacks are detailed in [Examples of current limitations that this RFC addresses](##examples-of-current-limitations-that-this-frc-addresses).
+The alternative is to add features to the provided type by alternative means such as inheriting from the provided type or by augmenting another type which defers to the provided type internally.  These drawbacks are detailed in [Examples of current limitations that this RFC addresses](#examples-of-current-limitations-that-this-rfc-addresses).
```
Comment (jsoref) [inline archived/FS-1061-Allow-intrinsic-type-extensions-for-provided-types.md:174]: I've removed a doubled `#` which I believe is incorrect for anchors as well as fixing the anchor ref...

File: tooling/FST-1033-analyzers.md [hunk 10]
```diff
@@ -141,7 +141,7 @@ simpler to start with `FSharp.Compiler.Analyzers.v1.0` only containing an API wh
 relevant logical source file contents**.
 
 This means analyzers that want to access the syntax tree would have to parse using their own private copy of FCS.
-Wach analyzer would have to run its own compilation/analysis internally, rather than having access to the FCS compiled trees.
+Each analyzer would have to run its own compilation/analysis internally, rather than having access to the FCS compiled trees.
```
Comment (jsoref) [inline tooling/FST-1033-analyzers.md:144]: My initial guess was `Watch`...

**Review by jsoref — COMMENTED** [review]
I understand these documents are mostly proposals, and I'm happy to drop anything that people want to keep things as they were originally written, but a quick check seems to indicate that documents are updated at later dates and I think that it's much easier to read content without typos and it's definitely easier to find things when things use the expected spelling of words...

**Review by T-Gro — APPROVED** [review]
I believe this is a good change for this repo.
When this is squashed, it should be also added to `.git-blame-ignore-revs` to exclude it from blaming history.

File: FSharp.Core-4.5.0.0/FS-1057-valueoption.md (hunk 7 above)
Comment (T-Gro) [reply FSharp.Core-4.5.0.0/FS-1057-valueoption.md:22]: squeeze

File: FSharp-6.0/FS-1087-resumable-code.md (hunk 5 above)
Comment (T-Gro) [reply FSharp-6.0/FS-1087-resumable-code.md:1027]: Using lowercase `nyi` is a common acronym, it should be kept.

File: FSharp-4.1/FS-1006-struct-tuples.md (hunk 2 above)
Comment (T-Gro) [reply FSharp-4.1/FS-1006-struct-tuples.md:297]: This line can be removed.

File: FSharp-4.0/StaticMethodArgumentsDesignAndSpec.md [hunk 11]
```diff
@@ -46,9 +46,9 @@ From the point of view of the type provider author, using the modified ProvideTy
       let m = ProvidedMethod("ExampleMethWithStaticParam", [ ], typeof<int>, IsStaticMethod = false)
       m.DefineStaticParameters(staticParams, (fun nm args ->
           let arg = args.[0] :?> int
-          let parms = [ for i in 1 .. arg -> ProvidedParameter("arg" + string i, typeof<int>)]
+          let params = [ for i in 1 .. arg -> ProvidedParameter("arg" + string i, typeof<int>)]
```
Comment (dsyme) [inline FSharp-4.0/StaticMethodArgumentsDesignAndSpec.md:49]: `params` is a reserved keyword in F#

**Review by dsyme — APPROVED** [review]

**Comment by dsyme** [comment]
Thank you!!!

File: FSharp-4.0/StaticMethodArgumentsDesignAndSpec.md (hunk 11 above)
Comment (jsoref) [reply FSharp-4.0/StaticMethodArgumentsDesignAndSpec.md:49]: @dsyme: did you want me to change this to `params_` or back to `parms`?

Comment (dsyme) [reply FSharp-4.0/StaticMethodArgumentsDesignAndSpec.md:49]: Thanks - I updated to `parameters` just now via direct commit

**Pull request merged**